README-Docker.md

```diff
@@ -0,0 +1,35 @@
+# Docker README
+
+A `Dockerfile` has been provided in order to make this code more convenient to run. To get started via `Docker`, you'll need:
+
+* A working Docker environment
+* A `git checkout` of this repository
+* A copy of `GeoLite2-Country.mmdb` in the root of this repository
+
+## Building Your Docker Container
+
+After you have checked out this repository and added `GeoLite2-Country.mmdb` to the root directory of the repository, run this command from the root directory:
+
+```bash
+docker build . -t mmdb-from-go
+```
+
+This will build your Docker container, build your `Go` code and also run the example code for you. To log in to your container:
+
+```bash
+docker run -it --volume $PWD:/project mmdb-from-go:latest /bin/bash
+```
+
+Once you have logged in, you can test out your freshly created database:
+
+```bash
+mmdbinspect -db GeoLite2-Country.mmdb \
```

Comment:
Somehow, this didn't work for me. I think the `&& ./mmdb-from-go-blogpost` part didn't run somehow; I had to run it manually once logged into the container -- then the `mmdbinspect` command worked.

```
nchelluri@grugbarn:~/dev/mmdb-from-go-blogpost [oalders/dockerfile=] $ sudo docker build . -t mmdb-from-go
[sudo] password for nchelluri: 
Sending build context to Docker daemon   7.65MB
Step 1/10 : FROM golang:buster
buster: Pulling from library/golang
756975cb9c7e: Pull complete 
d77915b4e630: Pull complete 
5f37a0a41b6b: Pull complete 
96b2c1e36db5: Pull complete 
145393847161: Pull complete 
3f8843661db9: Pull complete 
218d240e42d4: Pull complete 
Digest: sha256:ac6f3a6f28356b4a00d0a9c93c7d637eb9b7fac48ce78700ae2a862fc096592b
Status: Downloaded newer image for golang:buster
 ---> 8e2ffcb73e11
Step 2/10 : SHELL ["/bin/bash", "-o", "pipefail", "-c"]
 ---> Running in 7e5c7ce6e297
Removing intermediate container 7e5c7ce6e297
 ---> 88776cdbf224
Step 3/10 : ENV MMDBINSPECT_VERSION 0.1.1
 ---> Running in 8c9d39bc15a7
Removing intermediate container 8c9d39bc15a7
 ---> ccc9662dd0b5
Step 4/10 : ENV LESSCHARSET utf-8
 ---> Running in 69ac398c7513
Removing intermediate container 69ac398c7513
 ---> 45c85109ddb4
Step 5/10 : COPY . /project/
 ---> 6bd3f414f611
Step 6/10 : WORKDIR /tmp
 ---> Running in 317aeacc0b5f
Removing intermediate container 317aeacc0b5f
 ---> 6e7bea383aa6
Step 7/10 : RUN curl --location https://github.com/maxmind/mmdbinspect/releases/download/v0.1.1/mmdbinspect_${MMDBINSPECT_VERSION}_linux_amd64.deb -o mmdbinspect.deb && dpkg -i /tmp/mmdbinspect.deb
 ---> Running in d20ee1908f8c
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100   659  100   659    0     0   2804      0 --:--:-- --:--:-- --:--:--  2804
100  879k  100  879k    0     0  1427k      0 --:--:-- --:--:-- --:--:-- 1427k
Selecting previously unselected package mmdbinspect.
(Reading database ... 15410 files and directories currently installed.)
Preparing to unpack /tmp/mmdbinspect.deb ...
Unpacking mmdbinspect (0.1.1) ...
Setting up mmdbinspect (0.1.1) ...
Removing intermediate container d20ee1908f8c
 ---> e374a86b31cb
Step 8/10 : RUN apt-get update && apt-get install less
 ---> Running in b152f8cc2e54
Get:1 http://security.debian.org/debian-security buster/updates InRelease [65.4 kB]
Get:2 http://deb.debian.org/debian buster InRelease [121 kB]
Get:3 http://deb.debian.org/debian buster-updates InRelease [51.9 kB]
Get:4 http://security.debian.org/debian-security buster/updates/main amd64 Packages [254 kB]
Get:5 http://deb.debian.org/debian buster/main amd64 Packages [7907 kB]
Get:6 http://deb.debian.org/debian buster-updates/main amd64 Packages [7856 B]
Fetched 8408 kB in 2s (3436 kB/s)
Reading package lists...
Reading package lists...
Building dependency tree...
Reading state information...
The following NEW packages will be installed:
  less
0 upgraded, 1 newly installed, 0 to remove and 9 not upgraded.
Need to get 129 kB of archives.
After this operation, 288 kB of additional disk space will be used.
Get:1 http://deb.debian.org/debian buster/main amd64 less amd64 487-0.1+b1 [129 kB]
debconf: delaying package configuration, since apt-utils is not installed
Fetched 129 kB in 0s (810 kB/s)
Selecting previously unselected package less.
(Reading database ... 15416 files and directories currently installed.)
Preparing to unpack .../less_487-0.1+b1_amd64.deb ...
Unpacking less (487-0.1+b1) ...
Setting up less (487-0.1+b1) ...
debconf: unable to initialize frontend: Dialog
debconf: (TERM is not set, so the dialog frontend is not usable.)
debconf: falling back to frontend: Readline
Processing triggers for mime-support (3.62) ...
Removing intermediate container b152f8cc2e54
 ---> 37c1cb0fb24b
Step 9/10 : WORKDIR /project
 ---> Running in fa6ba86009ab
Removing intermediate container fa6ba86009ab
 ---> d82f47a12f33
Step 10/10 : RUN go build && ./mmdb-from-go-blogpost
 ---> Running in 7c5b889f9891
go: downloading github.com/maxmind/mmdbwriter v0.0.0-20200820202211-8086e3e6511b
go: downloading github.com/pkg/errors v0.9.1
go: downloading github.com/oschwald/maxminddb-golang v1.7.1-0.20200819192241-1f1e288ee3f9
go: downloading golang.org/x/sys v0.0.0-20191224085550-c709ea063b76
Removing intermediate container 7c5b889f9891
 ---> 0249d0c1347c
Successfully built 0249d0c1347c
Successfully tagged mmdb-from-go:latest
```

was the output from the previous command. I'm running with this Docker version:

```
nchelluri@grugbarn:~/dev/mmdb-from-go-blogpost [oalders/dockerfile=] $ docker -v
Docker version 19.03.13, build 4484c46d9d
```

Reply:
After some debugging with @apathor's help, I think that the think to do is remove the `--volume $PWD:/project` flag. I think that the files reside on the container but not in my host-dir. Do you agree?

Reply:
The purpose of that flag is to be able to edit your files on the host and have the changes mapped directly to the guest. Otherwise, you need to rebuild every time the code changes externally, so it would be nice to figure out exactly what's going on here.